llvm-bazel/WORKSPACE

```diff
@@ -3,9 +3,29 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 load(":configure.bzl", "llvm_configure")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
 
 llvm_configure(
     name = "llvm-project",
     overlay_path = "llvm-project-overlay",
     src_path = "../third_party/llvm-project",
 )
+
+maybe(
+    http_archive,
+    name = "vulkan_headers",
+    strip_prefix = "Vulkan-Headers-9bd3f561bcee3f01d22912de10bb07ce4e23d378",
+    sha256 = "19f491784ef0bc73caff877d11c96a48b946b5a1c805079d9006e3fbaa5c1895",
+    urls = [
+        "https://github.com/KhronosGroup/Vulkan-Headers/archive/9bd3f561bcee3f01d22912de10bb07ce4e23d378.tar.gz",
+    ],
+    build_file = "//third_party_build:vulkan_headers.BUILD",
+)
```

Comment:
Is the `maybe` here actually necessary?

It seems better to condition this on the Vulkan SDK being installed? Or does the `maybe` enable that?

If Bazel makes the above hard, just let me know.

Perhaps more interestingly -- could you sink this into the Vulkan SDK repository rule? Or into an adjacent one so that users that want Vulkan support don't have to replicate this rule to get the behavior?

Reply:
Sure, but my goal here is to not fetch the headers unless the SDK is actually available.

Basically -- if the SDK is available, you get the headers and the SDK. If not, you get neither.

Reply:
Ah you want to condition the headers on the SDK being installed? The headers are actually required for building, so that would make building this require the SDK, which doesn't seem right to me. Bazel is lazy on this: it only runs the repo rule if you try to build a thing that depends on it.

Reply:
I understand that it'll be lazy, but if i build //... and don't have the SDK installed, it'll still pull headers.

Maybe I'm to nervous about pulling ever more dependencies, but I am a bit nervous. ;]

Will it really *always* build without the SDK installed? Even link? Even in dynamic mode? I find this somewhat surprising....

Reply:
I think so? But I can't really guarantee. I'm mostly going off of the comments from people before me who actually understand what Vulkan is doing here :-D

I think a policy of "every target that transitively depends on an external dependency must be marked manual" would deal with your concern though, right? That's a bit annoying to enforce, although it's only like 5 targets right now, so probably nbd. Maybe bazel platform restrictions would help here, though I haven't played with them much.

My goal is that you should be able to at least *try* to build and query this target without the vulkan_sdk installed, which manual accomplishes. One of the things I don't like about Bazel selects is that they don't even let you try to do something wild. Like maybe this doesn't build for android, but I should be able to at least tell it to build for android explicitly and see what errors I get.

Reply:
There are really two issues here:
1. How do we disable unsupported rules?
2. What criteria should we use for disabling Vulkan rules?

For the first, I agree with you that tagging as manual is nicer (if it works) so that you *can* see what errors result even when the config isn't set up in the expected way.

For the second, I think it'd be good to be a bit more conservative in enabling rules with these kinds of dependencies. Specifically, if there *is* some system aspect necessary, I'd only enable them when that system aspect is present (and only then pull the archive of headers, etc.)

Reply:
I think for now tagging these as manual gives us something to work with that will enable integration with IREE so we can start dogfooding these there without making these less usable. We can continue to discuss the best way to handle this and other dependencies. SG?

Reply:
Sounds good.